### DISCUSSION TITLE
How the transition system in NER and Parser works.

### DISCUSSION THREAD

**Discussion by rkoystart** [explosion/spaCy#11791] [Help: Other Questions]
Hi, would like to know in detail about how the transition system works in NER and Parser. Have gone through this [video](https://www.youtube.com/watch?v=sqDHBH9IjRU&t=2605s). How things like how the state vector calculation, shift etc are not clear. Is there any other video or docs or blog on how it exactly works.

**Comment by polm** [comment]
We do not have detailed documentation of the parser implementation, the best reference is the code. However you might find these useful:

- [Parsing English in 500 Lines of Python](https://explosion.ai/blog/parsing-english-in-python) describes an early version of the spaCy parser in detail.
- Chapter 14 of [Speech and Language Processing](https://web.stanford.edu/~jurafsky/slp3/) covers shift-reduce parsers in general

Hope that helps.
